src/ocm/services/HostsService.ts

```diff
@@ -14,8 +14,7 @@ const HostsService = {
       const promises: AxiosPromise<void>[] = [];
       const hosts = await HostsService.listHostsBoundToCluster(clusterId);
       const hostDeleteParams = hosts.map<[string, string]>(({ infraEnvId, id }) => [
-        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-        infraEnvId!,
+        infraEnvId || '',
```

Comment:
How can we delete a host if we don't have the infraEnvId !?!?! :)

Reply:
We can't. :upside_down_face:

Reply:
I think the assumption was that [here](https://github.com/jtomasek/assisted-ui-lib/blob/26937b7e99ddd35b9cd13b1b4a5bf180a4b4a836/src/ocm/services/HostsService.ts#L15) we'll get for sure  a list of hosts for which all of them have the `infraEnvId` field set

Reply:
The problem is the same as with other places, TS thinks `infraEnvId` can be unset because of the API type definition.
https://github.com/openshift-assisted/assisted-ui-lib/blob/master/src/common/api/types.ts#L887

If we want to avoid having fallbacks where it doesn't make sense, maybe we can use `assert` .
`declare function assert(value: unknown): asserts value;`

Another solution would be to redefine the API types making the fields mandatory where we know they should be, so the code could use the updated type and not the base API type.
eg.  something like https://bobbyhadz.com/blog/typescript-make-property-required
```
type WithRequiredProperty<Type, Key extends keyof Type> = Type & {
  [Property in Key]-?: Type[Property];
}
```